runtime/interpreter/value.go

```diff
@@ -125,18 +125,23 @@ func (v TypeValue) String() string {
 	return fmt.Sprintf("Type<%s>", v.Type)
 }
 
-func (v TypeValue) Equal(other Value) BoolValue {
+func (v TypeValue) Equal(inter *Interpreter, other Value) BoolValue {
 	otherMetaType, ok := other.(TypeValue)
 	if !ok {
 		return false
 	}
-	return BoolValue(v.Type.Equal(otherMetaType.Type))
+
+	ty := inter.convertStaticToSemaType(v.Type)
+	otherTy := inter.convertStaticToSemaType(otherMetaType.Type)
+
+	return BoolValue(ty.Equal(otherTy))
```

Comment:
I understand that now checking the equality of two types need interpreter to convert types first.

I wonder you can think of a case where the result here would be different from the result of `BoolValue(v.Type.Equal(otherMetaType.Type))` (the previous result).

Reply:
It would not be any different, I just did not want to re-implement equality for the `StaticType` type hierarchy